### PR TITLE
Feature/add reindexing endpoint

### PR DESCRIPTION
**Pull request by SCBbestof**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -181,8 +186,8 @@ func main() {
 
 		// process and forward messages
 		pQConf := processor.NewProducerConfig(*kafkaProxyAddress, *combinedTopic, *kafkaProxyRoutingHeader)
+		forcedPQConf := processor.NewProducerConfig(*kafkaProxyAddress, *forcedCombinedTopic, *kafkaProxyRoutingHeader)
```
Comment (BerniVarga) [inline main.go]: Could you arrange the code in such a way that the related logical components are placed one next to the other? (The msg processor components, and the forced processor components - for readability purposes.)

File: processor/processor.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package processor
+
+import (
+	"github.com/Financial-Times/message-queue-go-producer/producer"
+	"github.com/Financial-Times/go-logger"
+	"encoding/json"
+)
+
+type Processor struct {
```
Comment (BerniVarga) [inline processor/processor.go]: Could we rename this as Forwarder? Or Producer? It's not a processor anymore, it only forwards the messages to Kafka

File: processor/processor.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+func (p *Processor) forwardMsg(headers map[string]string, model *CombinedModel) error {
+	// marshall message
+	b, err := json.Marshal(model)
+	if err != nil {
+		return err
+	}
+	// add special message type
+	headers["Message-Type"] = CombinerMessageType
```
Comment (BerniVarga) [inline processor/processor.go]: The constants that are used only in this file could be move here from the msg_processor file. Also, the construction of the Producer instance belongs here.

File: processor/forced_msg_processor.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+import (
+	"net/http"
+	"github.com/Financial-Times/go-logger"
+	"github.com/Financial-Times/post-publication-combiner/utils"
+	"github.com/dchest/uniuri"
+	"github.com/Financial-Times/message-queue-go-producer/producer"
+)
+
+type ForcedMsgProcessorI interface {
```
Comment (BerniVarga) [inline processor/forced_msg_processor.go]: Why do we use an interface here and none for the MsgProcessor? Does it make sense to use any interfaces at all in this context? Also, I have some problem with the naming. This is not actually a Msg Processor... it's more like a Publisher, Producer...

File: processor/processor.go (hunk 3 above)
Comment (BerniVarga) [reply processor/processor.go]: Similar happens in the forced_msg_processor file. Also, we might consider packaging differently. (optional)

File: main.go (hunk 1 above)
Comment (SCBbestof) [reply main.go]: Done

File: processor/processor.go (hunk 2 above)
Comment (SCBbestof) [reply processor/processor.go]: Good point. Renamed to Forwarder.

File: processor/processor.go (hunk 3 above)
Comment (SCBbestof) [reply processor/processor.go]: Not sure about the Producer. I changed the rest and added the producer construction in the main function, while cutting down arguments for the processors. Might be cleaner that way.

File: main.go [hunk 5]
```diff
@@ -180,24 +186,33 @@ func main() {
...
 
-		routeRequests(port, &requestHandler{processor: msgProcessor}, NewCombinerHealthcheck(msgProcessor.MsgProducer, mc.Consumer, &client, *docStoreAPIBaseURL, *publicAnnotationsAPIBaseURL))
+		// process requested messages - used for reindexing and forced requests
+		forcedPQConf := processor.NewProducerConfig(*kafkaProxyAddress, *forcedCombinedTopic, *kafkaProxyRoutingHeader)
+		forcedMsgProducer := producer.NewMessageProducerWithHTTPClient(forcedPQConf, &client)
+		requestProcessor := processor.NewRequestProcessor(
+			dataCombiner,
+			forcedMsgProducer,
+			*whitelistedContentTypes)
+		routeRequests(port, &requestHandler{requestProcessor: requestProcessor}, NewCombinerHealthcheck(requestProcessor.Forwarder.MsgProducer, mc.Consumer, &client, *docStoreAPIBaseURL, *publicAnnotationsAPIBaseURL))
```
Comment (BerniVarga) [inline main.go]: The healthcheck is currently verifying the connectivity to the producer and the consumer instances. In the background it is only doing a /topics call to the proxy, so, it makes no difference whether we are calling it for the consumer and producer instances. However, to keep the code clean and understandable, I would suggest whether 1) modifying the healthcheck message, reporting both of the queue names for the producer as well; 2) adding a separate check for all of the 4 topics.

Comment (BerniVarga) [reply main.go]: Yes, they were removed. For every check the /topics endpoint is called, but there is no verification that any topic can be found in the resultset, we only check for the status code of the response... (I don't like this solution, but yes, this is what we agreed on some time ago)

**Pull request merged**